quantum/send_string/send_string.c

```diff
@@ -156,44 +156,48 @@ void send_string_with_delay(const char *string, uint8_t interval) {
         if (!ascii_code) break;
         if (ascii_code == SS_QMK_PREFIX) {
             ascii_code = *(++string);
+
             if (ascii_code == SS_TAP_CODE) {
                 // tap
                 uint8_t keycode = *(++string);
                 tap_code(keycode);
+                wait_ms(interval);
             } else if (ascii_code == SS_DOWN_CODE) {
                 // down
                 uint8_t keycode = *(++string);
                 register_code(keycode);
+                wait_ms(interval);
             } else if (ascii_code == SS_UP_CODE) {
                 // up
                 uint8_t keycode = *(++string);
                 unregister_code(keycode);
+                wait_ms(interval);
             } else if (ascii_code == SS_DELAY_CODE) {
                 // delay
                 int     ms      = 0;
                 uint8_t keycode = *(++string);
+
                 while (isdigit(keycode)) {
                     ms *= 10;
                     ms += keycode - '0';
                     keycode = *(++string);
                 }
-                while (ms--)
-                    wait_ms(1);
+
+                wait_ms(ms);
```

Comment:
This can be moved down to eliminate the duplicate calls above.

Reply:
I actually did that first, but that made it so that on ASCII characters, it would wait twice: once during `send_char_with_delay` and then again when it returned to the call site.

Reply:
But that's in a separate branch. I'm not talking about where the loop was.

Reply:
Ohh, you mean at the end of the inner block, right! Good catch. Will change.

Reply:
Oh wait, actually, if we do this, then that means if there are any `SS_DELAY`s, it will wait that delay, and then the interval delay again. Is that okay? I guess that would make sense if the user did both an explicit delay as well as using `SEND_STRING_DELAY`

Reply:
If you want, you can rearrange this block to test for `SS_DELAY_CODE` first, then in an `else` do the rest, with the `wait_ms()` after.

Reply:
The more I think about this, the more I think it is better to have it the way it is. `SEND_STRING_DELAY` advertises itself as inserting a delay at the end of every key. It seems weird to break that by adding "unless you give an explicit delay action".

Though I don't feel strongly about it, so I am happy to make that change if you feel this should be the behavior.